tests/test_connection.py

```diff
@@ -24,16 +23,19 @@ def test_metric_matrix(self):
         result = self.connection.metric_matrix(base_point)
         expected = gs.array([gs.eye(self.dimension)])
 
-        gs.testing.assert_allclose(result, expected)
+        with self.session():
+            self.assertAllClose(gs.eval(result), gs.eval(expected))
```

Comment:
Could we put the `gs.eval` as part of the TestCase method? Would improve readability.